### DISCUSSION TITLE
PR to fix minor glitch in doc-examples

### DISCUSSION THREAD

**Discussion by hanseartic** [stellar/sorobanathon#29] [General]
https://github.com/stellar/soroban-docs/pull/208

Fixes IDE complaining about return type:
![image](https://user-images.githubusercontent.com/1752217/197525675-1600efff-4d8a-4770-b040-c1cc32c527dc.png)

**Comment by leighmcculloch** [comment]
As far as I'm aware this issue only appears to occur with IntelliJ.

It appears that the IDE is considering the `alloc` crate's prelude to be in scope for resolving macros. However, the file specifies `no_std` which explicitly excludes `alloc` and `std` preludes from being imported. Additionally there's an import of a `vec` macro that is being ignored.

I've opened this issue: https://youtrack.jetbrains.com/issue/IDEA-304495/nostd-is-ignored-when-resolving-macros

I think the only options we have in the SDK is to rename the `vec` macro and the `Vec` type to something else.

**Reply by hanseartic** [reply]
That makes perfect sense. I guess we can close the discussion at this place, then :)

**Comment by leighmcculloch** [comment]
IntelliJ uses rust-analyzer for macro expansion as an optional feature. @hanseartic If you enable macro expansion does that cause the error to go away?

**Reply by hanseartic** [reply]
thank you for the suggestion @leighmcculloch. unfortunately that does not work either.

**Comment by achallagundla** [comment]
Howdy Paul! 
Thank you for your submission will get back to you shortly after review!

**Comment by achallagundla** [comment]
Heya Paul,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of **250 XLM!** One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya
